doc/shape-defs.js

```diff
@@ -0,0 +1,52 @@
+"use strict";
+
+const path = require('path');
+const fs = require('fs');
+const packageJsonPath = path.resolve(__dirname, '../package.json');
+
+try {
+  const packageName = require(packageJsonPath).name;
+  const dtsPath = path.resolve(__dirname, `../dist/doc-temp/${packageName}.d.ts`);
+  let defs = fs.readFileSync(dtsPath).toString();
+
+  // aggregate external imports
+  const packages = {};
+  const importRegex = /^\s*import\s+\{([^}]+)\}\s*from\s*'([\w|-]+)'/gm;
+  let importMatch  = importRegex.exec(defs);
+  while (importMatch) {
+    const packageName = importMatch[2];
+    const imports = packages[packageName] || (packages[packageName] = []);
+    const bindings = importMatch[1].split(',').map(x => x.trim());
+    for (let binding of bindings) {
+      if (imports.indexOf(binding) === -1) {
+        imports.push(binding);
+      }
+    }
+    importMatch = importRegex.exec(defs);
+  }
+
+  // remove leading declare module
+  defs = defs.replace(/^declare module ".*" \{/, '');
```

Comment:
May I know why we need to do string replacement like this?

Reply:
@jdanyow might know.

Reply:
I see. So this is 100% based on validation. Sorry for not reading it carefully.  Then we cool here, if needed, i can adjust the build later. We kind of have better process/setup in other plugins repo after validation. Thanks @DrSammyD :+1:

Reply:
Sounds good.